[Content] Add a mission transporting a Hai that requires a Quantum Keystone (#9753)
endless-sky/endless-sky: data/hai/hai missions.txt

			label fine
			`	"You are being fined <fine> for abandoning your passengers without warning. You're lucky you didn't end up with a more serious punishment." After further chewing you out for endangering your passengers' safety, he walks away from you in a huff.`

mission "Hai Pilgrimage"
	name "Hai spiritual leader to <planet>"
	description "Bring a Hai spiritual leader to <destination>. He will pay you <payment>."
	minor
	clearance "You inform the authorities of the passenger you are transporting, and they quickly allow you to land."
	source
		government "Hai"

> 		not attributes "station"

Reviewer: The only thing I have left to say on this is that given the nature of the setting (an open air port and a climate capable of producing thick fog), it would be worth reading through the planet descriptions and either creating a list of possible offer locations, or at the very least excluding the most egregiously incompatible ones (i.e. Stonebreak, Darkmetal, etc.).

	destination

> 		attributes "unfettered"

Author: Seeing as this mission gives you 100 unfettered rep, I think the extra challenge of having to travel through unfettered space is justified.

> 		attributes "unfettered"

Reviewer 2: This looks like it can select planets in Wanderer space after the Unfettered Hai capture them. Is that something you want?

> 		attributes "unfettered"

Reviewer 3: Oh that'd be the only way to land on them lmao

> 		attributes "unfettered"

Author: > This looks like it can select planets in Wanderer space after the Unfettered Hai capture them. Is that something you want?

I think it's fine. Would fit into the whole 'big changes are coming' message Hareek has.

	passengers 1
	to offer
		random < 10
		or
			has "outfit: Quantum Keystone"
			has "outfit: Quantum Key Stone"
			has "outfit: Shield Beetle Pendant"

> 			has "outfit: Tree Skeleton Key Stone"

Author: He could just take a ride in one of your escorts, so I think it's fine if you have it installed on any ship.

		has "First Contact: Hai: offered"
	on offer
		conversation
			`It is an unusually foggy day on <origin>; the air is so thick with moisture that you cannot see more than a few paces in front of you. Perhaps that's why you're so startled when you turn around and see a Hai standing right next to you. He is so close you can see dewdrops on his whiskers, and you immediately notice the prominent jewelry that he is wearing. He appears to be quite old, judging by his stooped-over posture and the white fur around his muzzle.`

> 			branch normal
> 				has "outfit: Quantum Keystone"
> 			branch remnant
> 				has "outfit: Quantum Key Stone"
> 			branch pendant
> 				has "outfit: Shield Beetle Pendant"
> 			branch huh
> 				has "outfit: Tree Skeleton Key Stone"

Reviewer 4: ```suggestion
			branch pendant
				has "outfit: Shield Beetle Pendant"
			branch normal
				has "outfit: Quantum Keystone"
			branch remnant
				has "outfit: Quantum Key Stone"
			branch huh
				has "outfit: Tree Skeleton Key Stone"
```

Reviewer 4: This I think is the best hierarchy for multiple keystones.

			label normal

> 			`	"I see that you are wiser than most humans here," he says, pointing towards your ship. Its hatch is open, prominently displaying your Keystone to any potential travellers. "You carry a shield against the tempest of space. Let no one else tell you otherwise." He stops to catch his breath. "Now, onto more worldly things. My name is Hareek. Are you willing and able to take me to <destination> for <payment>?"`

Author: Kind of both?

> 			`	"I see that you are wiser than most humans here," he says, pointing towards your ship. Its hatch is open, prominently displaying your Keystone to any potential travellers. "You carry a shield against the tempest of space. Let no one else tell you otherwise." He stops to catch his breath. "Now, onto more worldly things. My name is Hareek. Are you willing and able to take me to <destination> for <payment>?"`

Reviewer 3: I mostly meant that you didnt say he was running or smth before
I mean doesnt matter much I was mostly curious

				goto choice
			label remnant
			`	"The keystone you carry is a strange one, human," he says, pointing towards your ship. Its hatch is open, prominently displaying your Keystone to any potential travellers. He walks up to it to get a closer look. "Very unusual. No proper mounting either - must be a human design." He sighs. "Well, at least the Keystone itself looks authentic. It will have to do." He turns to you. "You carry a shield against the tempest of space. Let no one else tell you otherwise. If your people have access to Keystones as well, then I hope they too become wise to its power, and learn to treat them with more reverence. Now, onto more worldly things. My name is Hareek. Are you willing and able to take me to <destination> for <payment>?"`
				goto choice
			label pendant

> 			`	"A unique design you have here, Captain," he says, pointing towards your ship. Its hatch is open, prominently displaying Nanachi's pendant to any potential travellers. "A gift, I presume? You are lucky you have someone looking out for you. You carry a shield against the tempest of space. Let no one else tell you otherwise." He stops to catch his breath. "Now, onto more worldly things. My name is Hareek. Are you willing and able to take me to <destination> for <payment>?"`

Reviewer 3: So the pendant isnt a smaller version of the quantum keystones? I thought u'd have it on you

			`	"That is an oversimplification, but it is not something I can explain adequately to a human." He turns away, looking into the fog. "There are portentous events occurring in the galaxy, and I must warn others of the turmoil I have foreseen. The Unfettered crave turmoil, so naturally they are more inclined to believe me." He gives a wry smile.`
			choice
				`	"Alright, I can take you there."`
					goto agree
				`	"Sorry, but this whole thing sounds a little too strange for me."`
					goto strange
				`	"Unfettered space is too dangerous for me. You'll have to find someone else."`
			label dangerous
			`	He snorts. "Dangerous? I am approaching 300 years of age, and yet I do not fear a trip to Unfettered space." He walks away, clearly disappointed.`
				decline
			label strange
			`	He sighs. "That's alright. I would not expect you to understand." He walks away, disappearing into the mist.`
				decline
			label agree
			`	"Excellent. Arriving in a human ship would be an auspicious start to my journey, but there are not many human captains who take proper precautions when traveling through space. I'm glad to see that you have some sense, at least." He lowers his voice.`

> 			`	"Since you seem to be a trustworthy captain, I will tell you of some things I have seen. Many years ago, I defended our space against the Korath raids. One day, we were caught unprepared and hit by a massive fleet. Every single ship in my patrol group was destroyed, except for ours - the only one with a keystone. I learned two things in my time with the Korath: that the old legends are true, and that the stability we have enjoyed as a species is ending. Changes in the galaxy are inevitable, and those changes are arriving soon." He begins to move toward your ship. "But let us not speak of this now. I have business to attend to."`

Reviewer 5: ```suggestion
			`	"Since you seem to be a trustworthy captain, I will tell you of some things I have seen. Many years ago, I defended our space against the Korath raids. One day, we were caught unprepared and hit by a massive fleet. Every single ship in my patrol group was destroyed, except for ours - the only one with a keystone. I learned two things in my time against the Korath: that the old legends are true, and that the stability we have enjoyed as a species is ending. Changes in the galaxy are inevitable, and those changes are arriving soon." He begins to move toward your ship. "But let us not speak of this now. I have business to attend to."`
```
`with` almost implied Hareek was with the Korath, not fighting against them. This word changes clarifies the intended meaning.

	description "Bring a Hai spiritual leader to <destination>."
	minor
	source
		government "Hai"
		not attributes "station" "unfettered"
	destination
		attributes "unfettered"
	passengers 1
	to offer
		random < 30
		has "outfit: Quantum Keystone"
		has "First Contact: Hai: offered"
	on offer
		conversation
			`It is an unusually foggy day on <source>. The air is so thick with moisture that you cannot see more than a few paces in front of you. Perhaps that is why you are so startled when you turn around and see a Hai standing right next to you. He is so close you can see dewdrops on his whiskers, and you immediately notice the prominent jewelry that he is wearing.`
			`	"I see that you are wiser than most humans here," he says, pointing towards your ship. "You carry a Keystone, a shield against the tempest of space. Let no one else tell you otherwise. Are you willing and able to take me to <destination>? I can pay <payment>." He appears quite old, judging by his stooped-over posture and the white fur around his muzzle.`
			choice
				`	"I can take you there."`
					goto agree
				`	"I'm not going there. You'll have to ask someone else."`
					decline

> 				`	"Why would you want to go to <destination>? Isn't that an Unfettered planet?"`

Reviewer 6: ```suggestion
				`	"Why would you want to go to <destination>? Isn't that an Unfettered planet?"`
					to display
						or
							has "visited system: Wah Ki"
							has "Wanderers: Translation Machine: offered"
							has "First Contact: Unfettered: offered"
```
This mission can offer when the player has never seen the Unfettered, making this conversation choice not make sense. You should only display it when the player has at least basic knowledge of the Unfettered. Alternatively, you can add these conditions to the `to offer` block instead.

			`	The Hai pauses for a moment, as if unsure of what to say. "My teachings are more... appreciated among our northern brethren."`
			choice
				`	"Teachings?"`
					goto teachings
				`	"Alright, I'll take you there."`
					goto agree
				`	"Unfettered space is too dangerous for me. You'll have to find someone else."`
					goto dangerous
			label teachings

> 			`	An unreadable expression appears on the Hai's face. "How do I explain this... You have your preachers and wise men among the humans, yes? I am something similar." He stops for a moment, realizing something. "Ah, apologies for my rudeness - you probably do not know who I am. My name is Hareek. Now, will you take me to <destination>?"`

Reviewer 6: ```suggestion
			`	An unreadable expression appears on the Hai's face. "How do I explain this... You have your preachers and wise men among the humans, yes? I am somewhat similar." He stops for a moment, realizing something. "Ah, apologies for my rudeness - you probably do not know who I am. My name is Hareek. Now, will you take me to <destination>?"`
```

> 			`	An unreadable expression appears on the Hai's face. "How do I explain this... You have your preachers and wise men among the humans, yes? I am something similar." He stops for a moment, realizing something. "Ah, apologies for my rudeness - you probably do not know who I am. My name is Hareek. Now, will you take me to <destination>?"`

Reviewer 6: I'm not sure I understand this bit of dialogue. Why would Hareek think that telling the player his name would make a difference?

Author: I was trying to imply here that Hareek is fairly well known among the Hai, and mistakenly assumed that the player would know who he is.

> 			`	An unreadable expression appears on the Hai's face. "How do I explain this... You have your preachers and wise men among the humans, yes? I am something similar." He stops for a moment, realizing something. "Ah, apologies for my rudeness - you probably do not know who I am. My name is Hareek. Now, will you take me to <destination>?"`

Reviewer 6: You can't only introduce Hareek by name in this conversation branch, because you use his name extensively in the `on complete` and `on visit` blocks. He should introduce himself no matter what route the player takes to the `accept`.

Author: I put his name in the intro to this mission, so this should be fixed now.

			choice
				`	"So you're going to preach to the Unfettered?"`
					goto preach
				`	"I can take you."`
					goto agree
				`	"Unfettered space is too dangerous for me. You'll have to find someone else."`
					goto dangerous
			label preach
			`	"That is an oversimplification, but it is not something I can explain adequately to a human." He turns away, looking into the fog. "There are portentious events occurring in the galaxy, and I must warn others of the turmoil I have forseen. The Unfettered crave turmoil, so naturally they are more inclined to believe me." He gives a wry smile.`
			choice
				`	"Alright, I can take you there."`
					goto agree
				`	"Sorry, but this whole thing sounds a little too strange for me."`
					goto strange
				`	"Unfettered space is too dangerous for me. You'll have to find someone else."`
			label dangerous

> 			`	He snorts. "Dangerous? I am approaching 300 years old, and yet I do not fear my own." He walks away, clearly disappointed.`

Reviewer 6: This part is a bit unclear. Is he calling the Unfettered "my own?" As in, "my own people?"

If so, I'm confused about his reaction. When he mentions his age, it's almost as if he's mocking the player for being scared of the Unfettered while an old Hai like himself is unafraid. But this is turned on its head when he refers to the Unfettered as "his own." He would have less reason to fear them, as they are "his own," so the insult falls flat.

Maybe you could have something like this instead
```suggestion
			`	He snorts. "Dangerous? I am approaching 300 years old, and yet I do not fear a trip to Unfettered space." He walks away, clearly disappointed.`
```
Or something along those lines that avoids the contradiction.

				decline
			label strange
			`	He sighs. "That's alright. I would not expect you to understand." He walks away, disappearing into the mist.`
				decline
			label agree

> 			`	"Excellent. There are not many captains nowadays, Hai or human, that take proper precautions when travelling through space. I'm glad to see that you have some sense, at least." He lowers his voice.`

Reviewer 6: ```suggestion
			`	"Excellent. There are not many captains nowadays, Hai or human, who take proper precautions when traveling through space. I'm glad to see that you have some sense, at least." He lowers his voice.`
```

> 			`	"Since you seem to be a trustworthy captain, I will tell you of some things I have seen. Many years ago, I defended our space against the Korath raids. We were hit by a massive fleet, and were caught unprepared. Every ship in my patrol group was destroyed, except for ours. Our ship was the only one with a Keystone. I learned two things in my time with the Korath - that the old legends are true, and that the stability we have enjoyed as a species is ending. Changes in the galaxy are inevitable, and those changes are arriving soon." He begins to move towards your ship. "But let us not speak of this now. I have business to attend to."`

Reviewer 6: ```suggestion
			`	"Since you seem to be a trustworthy captain, I will tell you of some things I have seen. Many years ago, I defended our space against the Korath raids. One day, we were hit by a massive fleet, and were caught unprepared. Every single ship in my patrol group was destroyed, except for ours - the only one with a Keystone. I learned two things in my time with the Korath: that the old legends are true, and that the stability we have enjoyed as a species is ending. Changes in the galaxy are inevitable, and those changes are arriving soon." He begins to move towards your ship. "But let us not speak of this now. I have business to attend to."`
```

				accept

	on visit
		dialog `You look for Hareek, but realize that he took a ride on one of your escorts! Better depart and wait for your escorts to arrive in this star system.`
	on complete
		payment 20000
		conversation

> 			`Hareek was quite tight-lipped about what he planned to do once on <destination>. The most you could get out of him was more stories about his past as a military pilot. Once you land, Hareek jumps off your ship, moving surprisingly fast for his age.`

Reviewer 6: ```suggestion
			`Hareek was quite tight-lipped about what he planned to do once on <planet>. The most you could get out of him was more stories about his past as a military pilot. Once you land, Hareek jumps off your ship, moving surprisingly fast for his age.`
```

> 			`"Thank you, captain, for making this journey. I apologize about not telling you more about me; the work I do is not something that I can reveal to any human who asks." He hands you <payment> and moves away. The surrounding Unfettered appear almost awed by his presence, and consequentially give you a wide berth.`

Reviewer 6: ```suggestion
			`"Thank you, Captain, for making this journey. I apologize for not telling you more about me; the work I do is not something that I can reveal to any human who asks." He hands you <payment> and moves away. The surrounding Unfettered appear almost awed by his presence and consequentially give you a wide berth.`
```

			label fine
			`	"You are being fined <fine> for abandoning your passengers without warning. You're lucky you didn't end up with a more serious punishment." After further chewing you out for endangering your passengers' safety, he walks away from you in a huff.`

mission "Hai Pilgrimage"
	name "Hai spiritual leader to <planet>"
	description "Bring a Hai spiritual leader to <destination>. He will pay you <payment>."
	minor
	source
		government "Hai"
		not attributes "station"
	destination
		attributes "unfettered"
	passengers 1
	to offer

> 		random < 30

Reviewer 3: That might be a bit too likely to trigger since it offers everywhere
10% maybe?

> 		has "outfit: Quantum Keystone"

Reviewer 7: Should this offer if you have the remnant key stone? Does the Hai know the difference?

Reviewer 8: This might be a good place to address the difference directly again with some branching and a defer, but maybe in another PR to hash out the specifics?

The campaign diplomat also recognizes the difference, iirc?

Reviewer 3: the description of the hai one explains the difference nowadays, its displayed in evidence or smth

Author: I have the feeling that Hareek would be able to tell the difference. I'll try adding a branch for if the player has a Remnant keystone installed instead.

Author: Added a branch for if the player has a Remnant keystone.

		has "First Contact: Hai: offered"
	on offer
		conversation

> 			`It is an unusually foggy day on <source>; the air is so thick with moisture that you cannot see more than a few paces in front of you. Perhaps that's why you're so startled when you turn around and see a Hai standing right next to you. He is so close you can see dewdrops on his whiskers, and you immediately notice the prominent jewelry that he is wearing. He appears to be quite old, judging by his stooped-over posture and the white fur around his muzzle.`

Reviewer 7: ```suggestion
			`It is an unusually foggy day on <origin>; the air is so thick with moisture that you cannot see more than a few paces in front of you. Perhaps that's why you're so startled when you turn around and see a Hai standing right next to you. He is so close you can see dewdrops on his whiskers, and you immediately notice the prominent jewelry that he is wearing. He appears to be quite old, judging by his stooped-over posture and the white fur around his muzzle.`
```

> 			`	"I see that you are wiser than most humans here," he says, pointing towards your ship. "You carry a Keystone, a shield against the tempest of space. Let no one else tell you otherwise. Are you willing and able to take me to <destination>? I can pay <payment>."`

Reviewer 3: Isnt this a bit too direct? Mb he's like that tho

Author: Inserted a pause before he asks you for transport.

			choice
				`	"I can take you there."`
					goto agree
				`	"I'm not going there. You'll have to ask someone else."`
					decline

> 				`	"Why would you want to go to <destination>? Isn't that an Unfettered planet?"`

Reviewer 3: I think you should move this option up so the last is decline and the first is accept

					to display
						or
							has "visited system: Wah Ki"
							has "Wanderers: Translation Machine: offered"
							has "First Contact: Unfettered: offered"

> 			`	The Hai pauses for a moment, as if unsure of what to say. "My teachings are more... appreciated among our northern brethren."`

Reviewer 8: ```suggestion
			`	The Hai pauses for a moment, as if unsure of what to say. "My teachings and prophecies are more... appreciated among our northern brethren."`
```

This or something like it would make the line in his log more justified.

			choice
				`	"Teachings?"`
				`	"Alright, I'll take you there."`
					goto agree
				`	"Unfettered space is too dangerous for me. You'll have to find someone else."`
					goto dangerous

> 			`	An unreadable expression appears on the Hai's face. "How do I explain this... You have your preachers and wise men among the humans, yes? I am somewhat similar." He stops for a moment, realizing something. "Ah, apologies for my rudeness - you probably do not know who I am. My name is Hareek. Now, will you take me to <destination>?"`

Reviewer 8: It's possible to skip this, so should be moved to or repeated in (he -is- old after all) the intro or accept string somewhere, unless we'd want to branch the later mentions?

Author: Should be fixed now.

			choice
				`	"So you're going to preach to the Unfettered?"`
				`	"I can take you."`
					goto agree
				`	"Unfettered space is too dangerous for me. You'll have to find someone else."`
					goto dangerous

> 			`	"That is an oversimplification, but it is not something I can explain adequately to a human." He turns away, looking into the fog. "There are portentous events occurring in the galaxy, and I must warn others of the turmoil I have forseen. The Unfettered crave turmoil, so naturally they are more inclined to believe me." He gives a wry smile.`

Reviewer 7: ```suggestion
			`	"That is an oversimplification, but it is not something I can explain adequately to a human." He turns away, looking into the fog. "There are portentous events occurring in the galaxy, and I must warn others of the turmoil I have foreseen. The Unfettered crave turmoil, so naturally they are more inclined to believe me." He gives a wry smile.`
```

			choice
				`	"Alright, I can take you there."`
					goto agree
				`	"Sorry, but this whole thing sounds a little too strange for me."`
					goto strange
				`	"Unfettered space is too dangerous for me. You'll have to find someone else."`
			label dangerous

> 			`	He snorts. "Dangerous? I am approaching 300 years old, and yet I do not fear a trip to Unfettered space." He walks away, clearly disappointed.`

Reviewer 7: ```suggestion
			`	He snorts. "Dangerous? I am approaching 300 years of age, and yet I do not fear a trip to Unfettered space." He walks away, clearly disappointed.`
```
It's an alien talking so I guess it's fine

> 			`	He snorts. "Dangerous? I am approaching 300 years old, and yet I do not fear a trip to Unfettered space." He walks away, clearly disappointed.`

Reviewer 8: It's going to get missed a lot, I think, since I'd expect most who refuse to not be invested enough to notice, but I absolutely adore what it says about his character that he blows of the danger of the Unfettered but won't travel on a ship without a keystone.

				decline
			label strange
			`	He sighs. "That's alright. I would not expect you to understand." He walks away, disappearing into the mist.`
				decline
			label agree
			`	"Excellent. There are not many captains nowadays, Hai or human, who take proper precautions when traveling through space. I'm glad to see that you have some sense, at least." He lowers his voice.`

> 			`	"Since you seem to be a trustworthy captain, I will tell you of some things I have seen. Many years ago, I defended our space against the Korath raids. One day, we were hit by a massive fleet, and were caught unprepared. Every single ship in my patrol group was destroyed, except for ours - the only one with a Keystone. I learned two things in my time with the Korath: that the old legends are true, and that the stability we have enjoyed as a species is ending. Changes in the galaxy are inevitable, and those changes are arriving soon." He begins to move towards your ship. "But let us not speak of this now. I have business to attend to."`

Reviewer 8: ```suggestion
			`	"Since you seem to be a trustworthy captain, I will tell you of some things I have seen. Many years ago, I defended our space against the Korath raids. One day, we were caught unprepared and hit by a massive fleet. Every single ship in my patrol group was destroyed, except for ours - the only one with a Keystone. I learned two things in my time with the Korath: that the old legends are true, and that the stability we have enjoyed as a species is ending. Changes in the galaxy are inevitable, and those changes are arriving soon." He begins to move towards your ship. "But let us not speak of this now. I have business to attend to."`
```

I'd also recommend swapping this order for both more dramatic effect and a subtle difference in Korath actions. With this change the fleet hits them when they can be caught unprepared, as it is now they are unprepared when the fleet hits them anyway.

				accept

	on visit

> 		dialog `You look for Hareek, but realize that he took a ride on one of your escorts! Better depart and wait for your escorts to arrive in this star system.`

Reviewer 3: He only gives you his name in the dangerous label

Reviewer 7: And if you ask about his teachings, but yeah, it's still not always known here. Nor in the `on complete` conversation.

Reviewer 8: If you ask his teachings, so still possible to skip over it.

Author: Should be fixed now, thanks for spotting this.

> 	on complete

Reviewer 3: How abput giving you 100 unfettered rep for bringing this legend?

Reviewer 8: Second this. If it's putting us up there safely, it should probably be helping us towards safely traveling there anyway, but a figure of this statute noting our wisdom for following Hai tradition would very specifically be something other Unfettered would notice.

Reviewer 3: Also would hopefully be a part of other missions to up uhai rep organically

Reviewer 7: On that note, shouldn't there be a reputation requirement for this mission to offer?

Reviewer 3: There's a choice option that says its dangerous to go there
I think this hai would not mind going in hostile territory either, he'd trust fate

Author: I think it would be pretty hard to get this mission organically if you needed to have both positive rep with the Unfettered and a keystone installed.
I added a rep bonus though, thanks for the suggestion.

		payment 20000
		conversation
			`Hareek was quite tight-lipped about what he planned to do once on <planet>. The most you could get out of him was more stories about his past as a military pilot. Once you land, Hareek jumps off your ship, moving surprisingly fast for his age.`
			`"Thank you, Captain, for making this journey. I apologize for not telling you more about me; the work I do is not something that I can reveal to any human who asks." He hands you <payment> and moves away. The surrounding Unfettered appear almost awed by his presence and consequentially give you a wide berth.`

> 		log "Minor People" "Hareek" `An old Hai who claims to foresee the future. Dropped him off in Unfettered space so he could spread his message there. He appears to be well-respected amongst the Unfettered.`

Reviewer 3: He claims to foresee the future? All he said is that change is coming, no?

Author: Ejothims suggested having him mention his prophecies alongside his teachings, so this should make more sense now.

			label fine
			`	"You are being fined <fine> for abandoning your passengers without warning. You're lucky you didn't end up with a more serious punishment." After further chewing you out for endangering your passengers' safety, he walks away from you in a huff.`

> mission "Pilgrimage"

Reviewer 9: ```suggestion
mission "Hai Pilgrimage"
```
Just make this slightly less generic.

	name "Hai spiritual leader to <planet>"

> 	description "Bring a Hai spiritual leader to <destination>."

Reviewer 9: ```suggestion
	description "Bring a Hai spiritual leader to <destination>. He will pay you <payment>."
```

	minor
	source
		government "Hai"
		not attributes "station"
	destination
		attributes "unfettered"
	passengers 1
	to offer
		random < 30
		has "outfit: Quantum Keystone"
		has "First Contact: Hai: offered"
	on offer
		conversation

> 			`It is an unusually foggy day on <source>. The air is so thick with moisture that you cannot see more than a few paces in front of you. Perhaps that is why you are so startled when you turn around and see a Hai standing right next to you. He is so close you can see dewdrops on his whiskers, and you immediately notice the prominent jewelry that he is wearing. He appears to be quite old, judging by his stooped-over posture and the white fur around his muzzle.`

Reviewer 9: ```suggestion
			`It is an unusually foggy day on <source>; the air is so thick with moisture that you cannot see more than a few paces in front of you. Perhaps that's why you're so startled when you turn around and see a Hai standing right next to you. He is so close you can see dewdrops on his whiskers, and you immediately notice the prominent jewelry that he is wearing. He appears to be quite old, judging by his stooped-over posture and the white fur around his muzzle.`
```
I dunno but this tone sounds a little funny. Not sure about my changes.

			`	"I see that you are wiser than most humans here," he says, pointing towards your ship. "You carry a Keystone, a shield against the tempest of space. Let no one else tell you otherwise. Are you willing and able to take me to <destination>? I can pay <payment>."`
			choice
				`	"I can take you there."`
					goto agree
				`	"I'm not going there. You'll have to ask someone else."`
					decline
				`	"Why would you want to go to <destination>? Isn't that an Unfettered planet?"`
					to display
						or
							has "visited system: Wah Ki"
							has "Wanderers: Translation Machine: offered"
							has "First Contact: Unfettered: offered"
			`	The Hai pauses for a moment, as if unsure of what to say. "My teachings are more... appreciated among our northern brethren."`
			choice
				`	"Teachings?"`

> 					goto teachings
> 				`	"Alright, I'll take you there."`
> 					goto agree
> 				`	"Unfettered space is too dangerous for me. You'll have to find someone else."`
> 					goto dangerous
> 			label teachings

Reviewer 9: ```suggestion
				`	"Alright, I'll take you there."`
					goto agree
				`	"Unfettered space is too dangerous for me. You'll have to find someone else."`
					goto dangerous
```
Pretty sure this is not necessary.

			`	An unreadable expression appears on the Hai's face. "How do I explain this... You have your preachers and wise men among the humans, yes? I am somewhat similar." He stops for a moment, realizing something. "Ah, apologies for my rudeness - you probably do not know who I am. My name is Hareek. Now, will you take me to <destination>?"`
			choice
				`	"So you're going to preach to the Unfettered?"`

> 					goto preach
> 				`	"I can take you."`
> 					goto agree
> 				`	"Unfettered space is too dangerous for me. You'll have to find someone else."`
> 					goto dangerous
> 			label preach

Reviewer 9: ```suggestion
				`	"I can take you."`
					goto agree
				`	"Unfettered space is too dangerous for me. You'll have to find someone else."`
					goto dangerous
```
Also think this is not necessary.

> 			`	"That is an oversimplification, but it is not something I can explain adequately to a human." He turns away, looking into the fog. "There are portentious events occurring in the galaxy, and I must warn others of the turmoil I have forseen. The Unfettered crave turmoil, so naturally they are more inclined to believe me." He gives a wry smile.`

Reviewer 9: ```suggestion
			`	"That is an oversimplification, but it is not something I can explain adequately to a human." He turns away, looking into the fog. "There are portentous events occurring in the galaxy, and I must warn others of the turmoil I have forseen. The Unfettered crave turmoil, so naturally they are more inclined to believe me." He gives a wry smile.`
```

	source
		government "Hai"
		not attributes "station"
	destination
		attributes "unfettered"
	passengers 1
	to offer
		random < 10
		or
			has "outfit: Quantum Keystone"
			has "outfit: Quantum Key Stone"
		has "First Contact: Hai: offered"
	on offer
		conversation
			`It is an unusually foggy day on <origin>; the air is so thick with moisture that you cannot see more than a few paces in front of you. Perhaps that's why you're so startled when you turn around and see a Hai standing right next to you. He is so close you can see dewdrops on his whiskers, and you immediately notice the prominent jewelry that he is wearing. He appears to be quite old, judging by his stooped-over posture and the white fur around his muzzle.`
			branch remnant
				has "outfit: Quantum Key Stone"
			`	"I see that you are wiser than most humans here," he says, pointing towards your ship. "You carry a Keystone, a shield against the tempest of space. Let no one else tell you otherwise." He stops to catch his breath. "Now, onto more worldly things. My name is Hareek. Are you willing and able to take me to <destination> for <payment>?"`
				goto choice
			label remnant

> 			`	"I see that you are wiser than most humans here," he says, pointing towards your ship. "You carry a Keystone, a shield against the tempest of space. Let no one else tell you otherwise." He stops to catch his breath. "I have noticed that the Keystone your ship bears is from beyond the wormhole. If your people have access to Keystones as well, then I hope they too become wise to its power. Now, onto more worldly things. My name is Hareek. Are you willing and able to take me to <destination> for <payment>?"`

Reviewer 3: I love the idea but is the remnant keystons in evidence enough for him to see it, or would he have to sense it?
@Zitchas what can you tell us?

Reviewer: The Hai should only react to their own Keystones, which come with certificates of authenticity and applicable display cases.
They should be ignorant of/suspicious of keystones from other sources.

Reviewer 10: Without getting into too much nitty gritty detail of how it where it's is mounted, the main difference is likely to be the fact that the Remnant use a different mount, and to a Hai it would look much more industrial and utilitarian than their own. Am interesting option might be if he accepts it, but expresses suspicion and concern that someone is making keystones without due reverence. Tinned with appreciation already included here. Sort of a "it is nice but..." Sorry if way. Maybe with encouragement to get a "proper" keystone, but accepting it for now.

What about if someone has one of each?

Reviewer 4: There are two other keystones:
Shield Beetle Pendant
Tree Skeleton Key Stone

Reviewer 3: Maybe the hai pretends to sense things but just uses a device to detect if there is a keystone or not?

Reviewer: My thinking is that the spaceport authorities keep a register of who does/does not have one, so that they can direct inquiries from applicably wealthy/superstitious individuals.
This being because I'd like to have luxury transport jobs on the job board (I don't think this exists yet?) that are exclusively available to captains with keystones.

Author: I added a bunch of branching options for the other kinds of keystones, and took into account Grey and Zitchas' feedback regarding Hareek's response to a Remnant keystone.
Frankly I have no idea what to do if someone has multiple kinds of keystones.

Author: I changed it so that it defaults to the Hai keystone dialog if you have that installed along with some other kind of keystone.

Reviewer 4: >  This being because I'd like to have luxury transport jobs on the job board (I don't think this exists yet?) that are exclusively available to captains with keystones.
The "transport wealthy passengers" mission requires both luxury accomodations and hai keystones.

			label choice
			choice
				`	"I can take you there."`
					goto agree
				`	"Why would you want to go to <destination>? Isn't that an Unfettered planet?"`
					goto ask
					to display
						or
							has "visited system: Wah Ki"
							has "Wanderers: Translation Machine: offered"
							has "First Contact: Unfettered: offered"
				`	"I'm not going there. You'll have to ask someone else."`
					decline
			label ask
			`	The Hai pauses for a moment, as if unsure of what to say. "My teachings and prophecies are more... appreciated among our northern brethren."`
			choice
				`	"Teachings?"`
				`	"Alright, I'll take you there."`
					goto agree
				`	"Unfettered space is too dangerous for me. You'll have to find someone else."`
					goto dangerous
			label teachings

> 			`	An unreadable expression appears on Hareek's face. "Ah, apologies for my rudeness - you probably do not know who I am. How do I explain this... You have your preachers and wise men among the humans, yes? I am something similar. Now, will you take me to <destination>?"`

Reviewer 3: ```suggestion
			`	An unreadable expression appears on Hareek's face. "Ah, apologies for my rudeness - you probably do not know who I am. How do I explain this... You have your preachers and wise men and women among the humans, yes? I am something similar. Now, will you take me to <destination>?"`
```
Thats more inclusive, right? Mb I'm the only one this slightly bugs

Reviewer 8: Best course for inclusiveness would be to "wise ones," and technically the real meaning is "wise humans" not "wise masculine humans," though there are still gendered connotations to the historic role which make the current language still apt.

If we do change this, "wise ones" or "wise humans" would be best; just repetition of binary genders has mostly phased out as an inclusive ideal, like "person first language."

Reviewer 3: I wanted to say wise humans but its "among the humans" so that'd be repetitive

			choice
				`	"So you're going to preach to the Unfettered?"`
				`	"I can take you."`
					goto agree
				`	"Unfettered space is too dangerous for me. You'll have to find someone else."`
					goto dangerous

> 			`	"That is an oversimplification, but it is not something I can explain adequately to a human." He turns away, looking into the fog. "There are portentous events occurring in the galaxy, and I must warn others of the turmoil I have foreseen. The Unfettered crave turmoil, so naturally they are more inclined to believe me." He gives a wry smile.`

Reviewer 3: Shouldnt this only trigger before the invasion? Like, I imagine he'd find his way there without you otherwise
Or mb just add that some already happened if its after that, and they need his council?

Author: I deliberately kept it vague as to what great event he was talking about, so that it could be applicable to Hai Reveal, the Wanderer campaign, the upcoming Unfettered campaign, etc.

Reviewer 8: That's a good point, that this could be read as applying to whatever -may- happen next, even after everything else directly affecting Unfettered in the game at any point in time is completed.

			choice
				`	"Alright, I can take you there."`
					goto agree
				`	"Sorry, but this whole thing sounds a little too strange for me."`
					goto strange
				`	"Unfettered space is too dangerous for me. You'll have to find someone else."`
			label dangerous
			`	He snorts. "Dangerous? I am approaching 300 years of age, and yet I do not fear a trip to Unfettered space." He walks away, clearly disappointed.`
				decline
			label strange

> 			`	He sighs. "That's alright. I would not expect you to understand." He walks away, disappearing into the mist.`

Reviewer 3: ~~how does he see the keystones in the mist to begin with~~

Reviewer 9: Perhaps he senses them.

Reviewer 3: that's be fine indeed but he says: `"I see that you are wiser than most humans here," he says, pointing towards your ship.`

Reviewer 8: "See" as a verb in English isn't always about literal sight, but "I sense" here would carry more weight for his appeal to mysticism, whether it's actually valid in universe or not.

Author: I just went with the simple, boring option and changed it so you have it mounted on your open hatch where he can see it.

				decline
			label agree

> 			`	"Excellent. There are not many captains nowadays, Hai or human, who take proper precautions when traveling through space. I'm glad to see that you have some sense, at least." He lowers his voice.`

Reviewer 3: Most hai have keystones

Reviewer: Better perhaps to have him say something about it being auspicious to arrive on a human vessel for handwavey reasons he won't explain.

Author: Changed the line taking into account Hurl and Grey's advice.

	source
		government "Hai"
		not attributes "uninhabited" "station"
		not planet "Darkmetal" "Dustmaker" "Stonebreak" "Skyfarm"
	destination
		attributes "unfettered"
	passengers 1
	to offer
		random < 10
		or
			has "outfit: Quantum Keystone"
			has "outfit: Quantum Key Stone"
			has "outfit: Shield Beetle Pendant"
			has "outfit: Tree Skeleton Key Stone"
		has "First Contact: Hai: offered"
	on offer
		conversation
			`It is an unusually foggy day on <origin>; the air is so thick with moisture that you cannot see more than a few paces in front of you. Perhaps that's why you're so startled when you turn around and see a Hai standing right next to you. He is so close you can see dewdrops on his whiskers, and you immediately notice the prominent jewelry that he is wearing. He appears to be quite old, judging by his stooped-over posture and the white fur around his muzzle.`
			branch pendant
				has "outfit: Shield Beetle Pendant"
			branch normal
				has "outfit: Quantum Keystone"
			branch remnant
				has "outfit: Quantum Key Stone"
			branch huh
				has "outfit: Tree Skeleton Key Stone"
			label normal

> 			`	"I see that you are wiser than most humans here," he says, pointing towards your ship. Its hatch is open, prominently displaying your Keystone to any potential travellers. "You carry a shield against the tempest of space. Let no one else tell you otherwise." He stops to catch his breath. "Now, onto more worldly things. My name is Hareek. Are you willing and able to take me to <destination> for <payment>?"`

Reviewer 6: ```suggestion
			`	"I see that you are wiser than most humans here," he says, pointing toward your ship. Its hatch is open, prominently displaying your keystone to any potential travelers. "You carry a shield against the tempest of space. Let no one else tell you otherwise." He pauses to take a breath. "Now, onto more worldly things. My name is Hareek. Are you willing and able to take me to <destination> for <payment>?"`
```
I think it makes more sense to make him just pause to breathe rather than play up the fact that he's winded.

				goto choice
			label remnant

> 			`	"The keystone you carry is a strange one, human," he says, pointing towards your ship. Its hatch is open, prominently displaying your Keystone to any potential travellers. He walks up to it to get a closer look. "Very unusual. No proper mounting either - must be a human design." He sighs. "Well, at least the Keystone itself looks authentic. It will have to do." He turns to you. "You carry a shield against the tempest of space. Let no one else tell you otherwise. If your people have access to Keystones as well, then I hope they too become wise to its power, and learn to treat them with more reverence. Now, onto more worldly things. My name is Hareek. Are you willing and able to take me to <destination> for <payment>?"`

Author: Dunno about the capitalization of 'keystone' here. I'll have to check.

> 			`	"The keystone you carry is a strange one, human," he says, pointing towards your ship. Its hatch is open, prominently displaying your Keystone to any potential travellers. He walks up to it to get a closer look. "Very unusual. No proper mounting either - must be a human design." He sighs. "Well, at least the Keystone itself looks authentic. It will have to do." He turns to you. "You carry a shield against the tempest of space. Let no one else tell you otherwise. If your people have access to Keystones as well, then I hope they too become wise to its power, and learn to treat them with more reverence. Now, onto more worldly things. My name is Hareek. Are you willing and able to take me to <destination> for <payment>?"`

Reviewer 6: It's mixed. The description of the outfit uses lowercase `keystone` (second paragraph), and I assume that's most relevant here, since he's not referring to the item in the Outfitter, but to the keystone itself.

<img width="281" alt="image" src="https://github.com/endless-sky/endless-sky/assets/98854112/21c52aff-31fb-4e66-8c20-ac0ff84f59c6">

> 			`	"The keystone you carry is a strange one, human," he says, pointing towards your ship. Its hatch is open, prominently displaying your Keystone to any potential travellers. He walks up to it to get a closer look. "Very unusual. No proper mounting either - must be a human design." He sighs. "Well, at least the Keystone itself looks authentic. It will have to do." He turns to you. "You carry a shield against the tempest of space. Let no one else tell you otherwise. If your people have access to Keystones as well, then I hope they too become wise to its power, and learn to treat them with more reverence. Now, onto more worldly things. My name is Hareek. Are you willing and able to take me to <destination> for <payment>?"`

Reviewer 9: #8355 quantum keystone is not capitalized, Quantum Key Stone is.

> 			`	"The keystone you carry is a strange one, human," he says, pointing towards your ship. Its hatch is open, prominently displaying your Keystone to any potential travellers. He walks up to it to get a closer look. "Very unusual. No proper mounting either - must be a human design." He sighs. "Well, at least the Keystone itself looks authentic. It will have to do." He turns to you. "You carry a shield against the tempest of space. Let no one else tell you otherwise. If your people have access to Keystones as well, then I hope they too become wise to its power, and learn to treat them with more reverence. Now, onto more worldly things. My name is Hareek. Are you willing and able to take me to <destination> for <payment>?"`

Author: Tibe mentioned that 'keystone' isn't capitalized outside of HR. Since 'keystone' seems to be the standard for most of the game, I think it should be lowercase along with 'key stone' for the same reason.

				goto choice
			label pendant
			`	"A unique design you have here, Captain," he says, pointing towards your ship. Its hatch is open, prominently displaying Nanachi's pendant to any potential travellers. "A gift, I presume? You are lucky you have someone looking out for you. You carry a shield against the tempest of space. Let no one else tell you otherwise." He stops to catch his breath. "Now, onto more worldly things. My name is Hareek. Are you willing and able to take me to <destination> for <payment>?"`
				goto choice
			label huh

> 			`	"I've never seen anything like this... thing you have," he says, pointing towards your ship. Its hatch is open, prominently displaying the Tree Skeleton Key Stone to any potential travellers. The Hai sounds more than a little disgusted. "I hope you haven't done anything you'll regret."`

Reviewer 6: ```suggestion
			`	"I've never seen anything like this... thing you have," he says, pointing toward your ship. Its hatch is open, prominently displaying the Tree Skeleton Key Stone to any potential travelers. The Hai sounds more than a little disgusted. "I hope you haven't done anything you'll regret."`
```

> 			`	He takes a closer look. "It seems to be made out of the same material as a Keystone, and yet it is entirely foreign to me. Where you got this, I have no idea. Well, at least the material itself looks authentic. It will have to do." He sighs and turns to you. "I suspect that whoever you got this from is not happy it is out of their hands. I would show more respect for Keystones if I were you." He stops to catch his breath. "Now, onto more worldly things. My name is Hareek. Are you willing and able to take me to <destination> for <payment>?"`

Reviewer 6: ```suggestion
			`	He takes a closer look. "It seems to be made out of the same material as a keystone, and yet it is entirely foreign to me. Where you got this, I have no idea. Well, at least the material itself looks authentic. It will have to do." He sighs and turns to you. "I suspect that whoever you got this from is not happy it is out of their hands. I would show more respect for keystones if I were you." He pauses to take a breath. "Now, onto more worldly things. My name is Hareek. Are you willing and able to take me to <destination> for <payment>?"`
```

			`	"That is an oversimplification, but it is not something I can explain adequately to a human." He turns away, looking into the fog. "There are portentous events occurring in the galaxy, and I must warn others of the turmoil I have foreseen. The Unfettered crave turmoil, so naturally they are more inclined to believe me." He gives a wry smile.`
			choice
				`	"Alright, I can take you there."`
					goto agree
				`	"Sorry, but this whole thing sounds a little too strange for me."`
					goto strange
				`	"Unfettered space is too dangerous for me. You'll have to find someone else."`
			label dangerous
			`	He snorts. "Dangerous? I am approaching 300 years of age, and yet I do not fear a trip to Unfettered space." He walks away, clearly disappointed.`
				decline
			label strange
			`	He sighs. "That's alright. I would not expect you to understand." He walks away, disappearing into the mist.`
				decline
			label agree
			`	"Excellent. Arriving in a human ship would be an auspicious start to my journey, but there are not many human captains who take proper precautions when traveling through space. I'm glad to see that you have some sense, at least." He lowers his voice.`

> 			`	"Since you seem to be a trustworthy captain, I will tell you of some things I have seen. Many years ago, I defended our space against the Korath raids. One day, we were caught unprepared and hit by a massive fleet. Every single ship in my patrol group was destroyed, except for ours - the only one with a Keystone. I learned two things in my time with the Korath: that the old legends are true, and that the stability we have enjoyed as a species is ending. Changes in the galaxy are inevitable, and those changes are arriving soon." He begins to move towards your ship. "But let us not speak of this now. I have business to attend to."`

Reviewer 6: ```suggestion
			`	"Since you seem to be a trustworthy captain, I will tell you of some things I have seen. Many years ago, I defended our space against the Korath raids. One day, we were caught unprepared and hit by a massive fleet. Every single ship in my patrol group was destroyed, except for ours - the only one with a keystone. I learned two things in my time with the Korath: that the old legends are true, and that the stability we have enjoyed as a species is ending. Changes in the galaxy are inevitable, and those changes are arriving soon." He begins to move toward your ship. "But let us not speak of this now. I have business to attend to."`
```
